inklecate: init at 1.0.0

shreerammodi opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

The Inklecate compiler does not exist in the nixpkgs repo. I (and 2.8k stargazers) find the ink language useful so adding its official compiler is sure to help some nix users

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"` *-- No packages depend on this change*
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Shrimpram commented:
The error is `The option lib.maintainers.shreerammodi.name' is used but not defined`. Could be because that's the latter commit, I'll try reordering them

shreerammodi committed:
maintainers: add shreerammodi

shreerammodi committed:
inklecate: init at version 1.0.0

inklecate is a command line compiler for the ink language, which is used
to make interactive narrative / choose-your-own-adventure style games.

More information about the ink language can be found at
https://github.com/inkle/ink

Shrimpram commented:
Let me know if it's merge-able and I'll squash

r-rmcgibbo commented:
Result of `nixpkgs-review pr 129591` at 95b6a047 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>inklecate</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/development/compilers/inklecate/default.nix:39:3:
```
   |
39 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

SuperSandro2000 commented:
> The error is `The option lib.maintainers.shreerammodi.name' is used but not defined`. Could be because the commit that updates the maintainers list is the latter commit, I'll try reordering them

That shouldn't matter.

SuperSandro2000 commented on pkgs/development/compilers/inklecate/default.nix:
```diff
@@ -0,0 +1,59 @@
+  pname = "inklecate";
+  version = "1.0.0";
+
+  repo = "https://github.com/inkle/ink";
+
+  src =
+    if stdenv.isLinux then
+      fetchurl {
+        url    = "${repo}/releases/download/v${version}/inklecate_linux.zip";
+        sha256 = "6e17db766222998ba0ae5a5da9857e34896e683b9ec42fad528c3f8bea7398ea";
+        name   = "${pname}-${version}";
+      }
+    else if stdenv.isDarwin then
+      fetchurl {
+        url    = "${repo}/releases/download/v${version}/inklecate_mac.zip";
```
```suggestion
  src =
    if stdenv.isLinux then
      fetchurl {
        url    = "https://github.com/inkle/ink/releases/download/v${version}/inklecate_linux.zip";
        sha256 = "6e17db766222998ba0ae5a5da9857e34896e683b9ec42fad528c3f8bea7398ea";
        name   = "${pname}-${version}";
      }
    else if stdenv.isDarwin then
      fetchurl {
        url    = "https://github.com/inkle/ink/releases/download/v${version}/inklecate_mac.zip";
```

Shrimpram replied:
Resolved by https://github.com/NixOS/nixpkgs/pull/129591/commits/7dd59fc311a3551925c745d605623e994bd08e9d

SuperSandro2000 commented on pkgs/development/compilers/inklecate/default.nix:
```diff
@@ -0,0 +1,59 @@
+
+  nativeBuildInputs = [ unzip makeWrapper ];
+
+  phases = [ "unpackPhase" "installPhase" ];
+
```
```suggestion
```

disable phases which should not run instead.

SuperSandro2000 commented on pkgs/development/compilers/inklecate/default.nix:
```diff
@@ -0,0 +1,59 @@
+    mkdir -p $out/bin/
+
+    cp $pname-$version/inklecate $out/bin/inklecate
+    '';
```
```suggestion
  '';
```

SuperSandro2000 commented on pkgs/development/compilers/inklecate/default.nix:
```diff
@@ -0,0 +1,59 @@
+
+    rm $pname-$version/ink-engine-runtime.dll
+    rm $pname-$version/ink_compiler.dll
+    '';
```
```suggestion
  '';
```

SuperSandro2000 commented on pkgs/development/compilers/inklecate/default.nix:
```diff
@@ -0,0 +1,59 @@
+      scripting language for writing interactive narrative
+      '';
+    homepage        = "https://www.inklestudios.com/ink/";
+    downloadPage    = "${repo}/releases";
```
```suggestion
    downloadPage    = "https://www.inklestudios.com/ink/";
```

SuperSandro2000 commented on pkgs/development/compilers/inklecate/default.nix:
```diff
@@ -0,0 +1,59 @@
+    maintainers     = with maintainers; [ shreerammodi ];
+  };
+}
+
```
One final new line to much.

Shrimpram replied:
![Screen Shot 2021-07-08 at 9 26 36 AM](https://user-images.githubusercontent.com/67710369/124958061-991cf180-dfce-11eb-8d9d-0e16af58d6d5.png)

SuperSandro2000 commented on pkgs/development/compilers/inklecate/default.nix:
```diff
@@ -0,0 +1,59 @@
+      '';
+    homepage        = "https://www.inklestudios.com/ink/";
+    downloadPage    = "${repo}/releases";
+    license         = lib.licenses.mit;
+    platforms       = platforms.linux ++ platforms.darwin;
```
```suggestion
    license         = licenses.mit;
    platforms       = platforms.unix;
```

Shrimpram commented:
So sorry for all the errors, a lot of expressions currently in master use such code, which is what I based a lot of my expression off.

SuperSandro2000 reviewed (changes requested):
Cna you please squash all the package related commits together? make sure to leave the maintainer one extra.
